### ISSUE TITLE
False positive in unsafe_vector_initialization: nonintegral vector element type

### ISSUE THREAD

**Issue by dtolnay** [rust-lang/rust-clippy#3477]
```rust
unsafe fn f<T: Copy, R: std::io::Read>(len: usize, io: R) -> Result<Vec<T>, ()> {
    let mut vec = Vec::<T>::with_capacity(len);
    vec.set_len(len);

    /* io.read_exact... */

    Ok(vec)
}
```

```console
error: unsafe vector initialization
 --> src/main.rs:3:5
  |
2 |     let mut vec = Vec::<T>::with_capacity(len);
  |                   ---------------------------- help: consider replace allocation with: `vec![0; len]`
3 |     vec.set_len(len);
  |     ^^^^^^^^^^^^^^^^
  |
```

The suggested replacement of `vec![0; len]` is not a viable replacement because the vector element type is not something that can be "0".

---

*`clippy 0.0.212 (b2601be 2018-11-27)`*

**Comment by oli-obk** [comment]
We should just stop emitting this lint for `set_len` entirely.

**Comment by dtolnay** [comment]
I deleted this lint in #3478.
